crates/precompiles/src/stablecoin_exchange/mod.rs

```diff
@@ -2997,1335 +2886,818 @@ mod tests {
-            path_usd
-                .token
-                .approve(
-                    alice,
-                    ITIP20::approveCall {
-                        spender: exchange.address,
-                        amount: U256::from(min_order_amount * 10),
-                    },
-                )
-                .expect("Failed to approve PathUSD");
-        }
-
-        // Place orders to provide liquidity at 1:1 rate (tick 0)
-        // For trade USDC -> PathUSD -> EURC:
-        // - First hop needs: bid on USDC (someone buying USDC with PathUSD)
-        // - Second hop needs: ask on EURC (someone selling EURC for PathUSD)
-
-        // USDC bid: buy USDC with PathUSD
-        exchange
-            .place(alice, usdc_addr, min_order_amount * 5, true, 0)
-            .expect("Failed to place USDC bid order");
-
-        // EURC ask: sell EURC for PathUSD
-        exchange
-            .place(alice, eurc_addr, min_order_amount * 5, false, 0)
-            .expect("Failed to place EURC ask order");
-
-        exchange
-            .execute_block(Address::ZERO)
-            .expect("Failed to execute block");
-
-        // Quote multi-hop: USDC -> PathUSD -> EURC
-        let amount_in = min_order_amount;
-        let amount_out = exchange
-            .quote_swap_exact_amount_in(usdc_addr, eurc_addr, amount_in)
-            .expect("Should quote multi-hop trade");
-
-        // With 1:1 rates at each hop, output should equal input
-        assert_eq!(
-            amount_out, amount_in,
-            "With 1:1 rates, output should equal input"
-        );
+        StorageCtx::enter(&mut storage, || {
+            let mut exchange = StablecoinExchange::new();
+            exchange.initialize()?;
+
+            let admin = Address::random();
+            let alice = Address::random();
+            let min_order_amount = MIN_ORDER_AMOUNT;
+            let min_order_amount_x10 = U256::from(MIN_ORDER_AMOUNT * 10);
+
+            // Setup: PathUSD <- USDC
+            //        PathUSD <- EURC
+            let _path_usd = TIP20Setup::path_usd(admin)
+                .with_issuer(admin)
+                .with_mint(alice, min_order_amount_x10)
+                .with_approval(alice, exchange.address, min_order_amount_x10)
+                .apply()?;
+            let usdc = TIP20Setup::create("USDC", "USDC", admin)
+                .with_issuer(admin)
+                .with_mint(alice, min_order_amount_x10)
+                .with_approval(alice, exchange.address, min_order_amount_x10)
+                .apply()?;
+            let eurc = TIP20Setup::create("EURC", "EURC", admin)
+                .with_issuer(admin)
+                .with_mint(alice, min_order_amount_x10)
+                .with_approval(alice, exchange.address, min_order_amount_x10)
+                .apply()?;
+
+            // Create pairs
+            exchange.create_pair(usdc.address())?;
+            exchange.create_pair(eurc.address())?;
```

Comment:
nit: we no longer need to create pairs, this will happen automatically when calling place if the pair does not exist.